README.md

```diff
@@ -1,26 +1,61 @@
-<!--
-Avoid using this README file for information that is maintained or published elsewhere, e.g.:
+# namecheap-lego-operator
 
-* metadata.yaml > published on Charmhub
-* documentation > published on (or linked to from) Charmhub
-* detailed contribution guide > documentation or CONTRIBUTING.md
+## Description
 
-Use links instead.
--->
+Namecheap LEGO operator implements the provider side of the `tls-certificates-interface`
+to provide signed certificates from an ACME servers, using LEGO
+(https://go-acme.github.io/lego).
+It uses the  `acme_client` library to get the certificate from the ACME server.
 
-# namecheap-lego-operator
+## Config
+
+### Required configuration properties
+```
+email: <Account email address>
+domain: <Domain to request certificate for>
+namecheap-username: <Namecheap API user>
+namecheap-api-key: <Namecheap API key>
+```
+
+### Optional configuration properties
+```
+namecheap-http-timeout: <API request timeout in seconds>
+namecheap-polling-interval: <Time between DNS propagation checks in seconds>
+namecheap-propagation-timeout: <Maximum waiting time for DNS propagation in seconds>
+namecheap-ttl: <The TTL of the TXT record used for the DNS challenge>
+namecheap-sandbox: <Use Namecheap sandbox API>
+```
+## Usage
```

Comment:
I'd place the `usage` before the `config` section. It's the first thing users will be interested in.

Reply:
✅